src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -553,6 +553,82 @@ static Value builtinStringReplace(ExecutionState& state, Value thisValue, size_t
     }
 }
 
+static Value builtinStringReplaceAll(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (thisValue.isUndefinedOrNull()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().object.string(), true, state.context()->staticStrings().replace.string(), ErrorObject::Messages::GlobalObject_ThisUndefinedOrNull);
+    }
+    Value searchValue = argv[0];
+    Value replaceValue = argv[1];
+    // If searchValue is neither undefined nor null, then
+    if (!searchValue.isUndefinedOrNull()) {
+        bool isSearchValueRegExp = searchValue.isObject() && searchValue.asObject()->isRegExp(state);
+        // If isRegExp is true, then
+        if (isSearchValueRegExp) {
+            String* flags = searchValue.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().flags)).value(state, searchValue).toString(state);
+            if (flags->find("g") == SIZE_MAX) {
+                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().String.string(), true, state.context()->staticStrings().match.string(), ErrorObject::Messages::GlobalObject_ThisUndefinedOrNull);
+            }
+        }
+        // Let replacer be ? GetMethod(searchValue, @@replace).
+        Value replacer = Object::getMethod(state, searchValue, ObjectPropertyName(state.context()->vmInstance()->globalSymbols().replace));
+        if (!replacer.isUndefined()) {
+            Value args[2] = { thisValue, replaceValue };
+            // Return ? Call(replacer, searchValue, « O, replaceValue »).
+            return Object::call(state, replacer, searchValue, 2, args);
+        }
+    }
+
+    String* string = thisValue.toString(state);
+    String* searchString = searchValue.toString(state);
+    String* replaceString = nullptr;
+    bool functionalReplace = replaceValue.isCallable();
+    // If functionalReplace is false, then
+    if (!functionalReplace) {
+        replaceString = replaceValue.toString(state);
+    }
+    size_t advanceBy = 1;
+    size_t searchLength = searchString->length();
+
+    // Let advanceBy be max(1, searchLength).
+    if (advanceBy < searchLength) {
+        advanceBy = searchLength;
+    }
+    std::vector<size_t> matchPositions;
+    // Let position be ! StringIndexOf(string, searchString, 0).
+    size_t position = string->find(searchString, 0);
+
+    // Repeat, while position is not -1
+    while (position != SIZE_MAX) {
+        matchPositions.push_back(position);
+        position = string->find(searchString, position + advanceBy);
+    }
+    size_t endOfLastMatch = 0;
+    size_t matchPositionsSize = matchPositions.size();
+    StringBuilder builder;
+    String* replacement;
+    // For each element p of matchPositions, do
+    for (uint i = 0; i < matchPositionsSize; i++) {
```

Comment:
How about replacing `matchPositionsSize` variable with `matchPositions.size()` ?
`matchPositionsSize` variable is used only for this for statement.